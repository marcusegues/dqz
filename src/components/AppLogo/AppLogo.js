// @flow
import React from 'react';
// $FlowFixMe
import { View, Image } from 'react-native';
import { moderateScale } from '../../styles/Scaling';

const logo = require('../../../assets/images/quickZollLogo.png');
const logoWithShadow = require('../../../assets/images/quickZollLogoWithShadow.png');

const ownStyles = {
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    width: moderateScale(193),
    height: moderateScale(135),
  },
  logoWithShadow: {
    width: moderateScale(205),
    height: moderateScale(160),
  },
};

type AppLogoProps = {
  withShadow?: boolean,
};

export const AppLogo = ({ withShadow }: AppLogoProps) => (
  <View style={ownStyles.container}>
    <Image
      source={withShadow ? logoWithShadow : logo}
      style={withShadow ? ownStyles.logoWithShadow : ownStyles.logo}
      resizeMode="contain"
    />
  </View>
);

AppLogo.defaultProps = {
  withShadow: false,
};
