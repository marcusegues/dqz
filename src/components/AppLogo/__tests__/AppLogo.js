import renderer from 'react-test-renderer';
import React from 'react';
import { AppLogo } from '../AppLogo';

describe('AppLogo', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<AppLogo />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
