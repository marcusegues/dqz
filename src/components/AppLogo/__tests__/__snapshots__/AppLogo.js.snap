// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLogo renders according to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 212.14285714285714,
        "width": 303.2857142857143,
      }
    }
  />
</View>
`;
